src/elements/OcSidebar.vue

```diff
@@ -0,0 +1,221 @@
+
+  components: {
+    OcSidebarNavItem,
+    OcImg,
+    OcButton,
+    OcIcon
+  },
+
+  props: {
+    /**
+     * Source of the logo image
+     */
+    logoImg: {
+      type: String,
+      required: false
+    },
+    /**
+     * Name of the product where the sidebar is used
+     */
+    productName: {
+      type: String,
+      required: true
+    },
+    /**
+     * Navigation items of the product
+     */
+    navItems: {
+      type: Array,
+      required: true
+    },
+    /**
+     * Asserts whether the sidebar's position is fixed
+     */
+    fixed: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    /**
+     * Accessibility label of the close button
+     */
+    closeButtonLabel: {
+      type: String,
+      required: false,
+      default: "Close navigation menu"
+    }
+  },
+
+  computed: {
+    $_ocSidebar_classes() {
+      const classes = ["oc-sidebar"]
+
+      if (this.fixed) {
+        classes.push("oc-sidebar-fixed")
+      }
+
+      return classes
+    }
+  },
+
+  methods: {
+    $_ocSidebar_buttonClose_click() {
+      /**
+       * The user clicked on the close button
+       */
+      this.$emit("close")
+    }
+  }
+}
+</script>
+
+<docs>
+Sidebar component containing branding and navigation
+
+## Examples
+```jsx
+  <template>
+    <div>
+      <oc-sidebar
+        logoImg="https://owncloud.org/wp-content/themes/owncloud/img/owncloud-org-logo.svg"
+        productName="ownCloud"
+        :navItems="navItems"
+      />
+    </div>
+  </template>
+  <script>
+    export default {
+      computed: {
+        navItems() {
+          return [
+            { name: 'Home', route: { path: '/' }, icon: 'home', active: true },
+            { name: 'All files', route: { path: '/files' }, icon: 'folder' },
+            { name: 'Shared files', route: { path: '/shared' }, icon: 'share' }
+          ]
+        }
+      }
+    }
+  </script>
+```
+
+If a source of the logo image is not provided, the product name is used instead.
+
+```jsx
+  <template>
+    <div>
+      <oc-sidebar
+        productName="ownCloud"
+        :navItems="navItems"
+      />
+    </div>
+  </template>
+  <script>
+    export default {
+      computed: {
+        navItems() {
+          return [
+            { name: 'Home', route: { path: '/' }, icon: 'home' },
+            { name: 'All files', route: { path: '/files' }, icon: 'folder', active: true },
+            { name: 'Shared files', route: { path: '/shared' }, icon: 'share' }
+          ]
+        }
+      }
+    }
+  </script>
+```
+
+To provide additional content above the navigation or in the footer, you can use the `upperContent` and `footer` slots.
+The navigation block will automatically receive bottom margin in case the `footer` slot exists.
+
+```jsx
+  <template>
+    <div>
+      <oc-sidebar
+        logoImg="https://owncloud.org/wp-content/themes/owncloud/img/owncloud-org-logo.svg"
+        productName="ownCloud"
+        :navItems="navItems"
```

Comment:
Sure, sounds good 👍

Reply:
I'll overwrite the sidebar height in the examples via utility class. Setting the height globally would lead to a need for scrolling then in them which feels a bit odd and setting it only locally also doesn't feel right.